mllib/src/main/scala/org/apache/spark/mllib/evaluation/MultilabelMetrics.scala

```diff
@@ -142,17 +124,17 @@ class MultilabelMetrics @Since("1.2.0") (predictionAndLabels: RDD[(Array[Double]
     if((p + r) == 0) 0.0 else 2 * p * r / (p + r)
   }
 
-  private lazy val sumTp = tpPerClass.foldLeft(0L) { case (sum, (_, tp)) => sum + tp }
-  private lazy val sumFpClass = fpPerClass.foldLeft(0L) { case (sum, (_, fp)) => sum + fp }
-  private lazy val sumFnClass = fnPerClass.foldLeft(0L) { case (sum, (_, fn)) => sum + fn }
+  private lazy val sumTp = summary.tpPerClass.values.sum
+  private lazy val sumFpClass = summary.fpPerClass.values.sum
+  private lazy val sumFnClass = summary.fnPerClass.values.sum
 
   /**
    * Returns micro-averaged label-based precision
    * (equals to micro-averaged document-based precision)
    */
   @Since("1.2.0")
   lazy val microPrecision: Double = {
-    val sumFp = fpPerClass.foldLeft(0L) { case(cum, (_, fp)) => cum + fp}
+    val sumFp = summary.fpPerClass.foldLeft(0L) { case(cum, (_, fp)) => cum + fp}
```

Comment:
Same point about .values.sum here I think

Reply:
@zhengruifeng maybe we should change this too for completeness, but I'll merge later today either way

Reply:
@srowen  I think I can change this place in MultilabelEvaluator